euphoria-avro/src/main/java/cz/seznam/euphoria/avro/AvroHadoopSource.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016-2018 Seznam.cz, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.seznam.euphoria.avro;
+
+import cz.seznam.euphoria.hadoop.input.HadoopFileSource;
+import org.apache.avro.generic.GenericData.Record;
+import org.apache.avro.mapred.AvroKey;
+import org.apache.avro.mapreduce.AvroKeyInputFormat;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+
+/**
+ * Read data from Avro format
+ */
+public class AvroHadoopSource extends HadoopFileSource<AvroKey<Record>, NullWritable> {
+
+  private static final long serialVersionUID = -714547320436003072L;
+
+  /**
+   * @param path file input path, could be file://, hdfs:// or any other schema
+   * @param conf standard Hadoop configuration
+   * @param schema read from JSON file
+   */
+  @SuppressWarnings({ "unchecked", "rawtypes" })
+  public AvroHadoopSource(Path path, Configuration conf, String schema) {
+    super((Class) AvroKey.class, NullWritable.class, AvroKeyInputFormat.class,
+            updateConf(path, conf, schema));
+  }
+
+  public AvroHadoopSource(Path path, String schema) {
+    this(path, new Configuration(), schema);
+  }
+
+  public AvroHadoopSource(Path path) {
+    this(path, new Configuration(), null);
+  }
+
+  private static Configuration updateConf(Path path, Configuration conf, String schema) {
```

Comment:
If we have optional parameters, it would be better to create a builder, so the user knows what parameters source accepts (you can take a look at `SequenceFileSink.of(..)`).